src/main/java/monobot/util/TaskList.java

```diff
@@ -0,0 +1,116 @@
+package monobot.util;
+
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+import monobot.task.Task;
+
+/**
+ * Manages a ArrayList of Task objects.
+ * Contains methods to add, delete, mark, unmark and retrieve tasks in the list.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructs a new TaskList object
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a new TaskList object with an ArrayList of tasks
+     *
+     * @param tasks ArrayList containing Task objects
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Adds a new task to the list.
+     *
+     * @param task Task to be added to the list.
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Deletes a task from the list.
+     *
+     * @param index index of Task in list to be deleted.
+     */
+    public void deleteTask(int index) {
+        tasks.remove(index);
+    }
+
+    /**
+     * Gets a task from the list.
```

Comment:
I think its great that you're very thorough in all your javadocs comments! Although would it make the code unnecessarily lengthy to document getters/setters in your code as well?